From 7a375cc6c561b4f9886d4ecaf8cda3abb959ac1e Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Wed, 3 Jul 2024 11:59:48 +0200
Subject: [PATCH 0990/1695] drivers: clk-stm32-rcc: rely on DT for initial MCO
 pin secure state

Remove call to stm32_pinctrl_set_secure_cfg() as we should rely on the
device tree for the pin secure state at platform initialization.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 66589cc6ab328eaace3b782e8becf0cfea99e945 drivers: clk-stm32-rcc: rely on DT for initial MCO pin secure state]
Change-Id: I9500a156c56599fe26406fec0eafa84611945f52
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/391137
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/drivers/clk/clk-stm32-rcc.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/core/drivers/clk/clk-stm32-rcc.c b/core/drivers/clk/clk-stm32-rcc.c
index e41622d78..02e90561c 100644
--- a/core/drivers/clk/clk-stm32-rcc.c
+++ b/core/drivers/clk/clk-stm32-rcc.c
@@ -64,7 +64,6 @@ static TEE_Result stm32mp_rcc_probe(const void *fdt, int node,
 	}
 
 	if (rcc_pinctrl) {
-		stm32_pinctrl_set_secure_cfg(rcc_pinctrl, true);
 		res = pinctrl_apply_state(rcc_pinctrl);
 		if (res)
 			return res;
-- 
2.47.2

